api/core/types.proto

```diff
@@ -28,6 +28,7 @@ message Condition {
     string reason = 3;
     string message = 4;
     string timestamp = 5;
+    bool suspend = 6;
```

Comment:
I wouldn't expect this to get treated like a condition. I would instead add it as a bool directly on the `GitRepository` type.